src/index.ml

```diff
@@ -66,7 +66,49 @@ let src = Logs.Src.create "index" ~doc:"Index"
 
 module Log = (val Logs.src_log src : Logs.LOG)
 
+module IndexMetrics = struct
```

Comment:
that case is a not conventional in OCaml lands :-) `Index_metrics` ?

Reply:
Indeed :wink: I actually don't like this being in the same file at all; I could define these metrics in a separate `index_metrics.ml` unit, along with the IO instrementation functor. Thoughts?

Reply:
sounds better indeed